polkadot/xcm/xcm-builder/src/barriers.rs

```diff
@@ -52,7 +52,7 @@ impl ShouldExecute for TakeWeightCredit {
 	}
 }
 
-const MAX_ASSETS_FOR_BUY_EXECUTION: usize = 1;
+const MAX_ASSETS_FOR_BUY_EXECUTION: usize = 5;
```

Comment:
just wonder (maybe I am missing something), if it is good idea to expose and make it runtime-configurable like:
```
pub struct AllowTopLevelPaidExecutionFrom<T, MaxAssetsForBuyExecution>(PhantomData<(T, MaxAssetsForBuyExecution)>);

impl<
    T: Contains<MultiLocation>,
    MaxAssetsForBuyExecution: Get<u8>
> ShouldExecute for AllowTopLevelPaidExecutionFrom<T> {
```

Reply:
Let's already do this if this needs to be 5 only in asset-hub?

Reply:
well, the update should really be universal. all the chains that supports multiple assets needs this.

Reply:
No, not that support multiple asset. That support _buying execution_ in multiple assets.

Asset Hub does not support this.

Reply:
You can see how this works now, using a [Take First Asset Trader](https://github.com/paritytech/polkadot-sdk/blob/bdbe982/cumulus/primitives/utility/src/lib.rs#L120). It (conveniently) allows you to put `MAX_ASSETS` into holding, but only takes the first one for `BuyExecution`. So this is still the same behavior; it allows 1 asset for `BuyExecution`.

Again, I know this PR might solve the short-term problem for existing use cases, but I think the intent of this `const` is clear and that a chain supporting multiple assets does not mean it supports multiple assets simultaneously for weight payment.

Reply:
```suggestion
const MAX_ASSETS_FOR_BUY_EXECUTION: usize = 2;
```